include/openthread-config-android.h

```diff
@@ -47,21 +41,6 @@
 /* Define to 1 if you want to use child supervision feature */
 #define OPENTHREAD_ENABLE_CHILD_SUPERVISION 1
 
-/* Define to 1 to enable the commissioner role. */
-#define OPENTHREAD_ENABLE_COMMISSIONER 1
```

Comment:
This PR lets us define a new config file when we need it which is very useful :+1: 

Suggestion: Should we treat this file then as the default (for travis build) and consider keeping most of features enabled to make sure they are covered by  travis andriod build. Thought?

Reply:
Agree! I have re-enabled these optional features in Android.mk if its built with `BUILD_PRODUCT=generic`. `BUILD_PRODUCT` will be set to a real name on real product environment.